numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -3423,26 +3431,54 @@ PyArray_Where(PyObject *condition, PyObject *x, PyObject *y)
         /* Get the result from the iterator object array */
         ret = (PyObject*)NpyIter_GetOperandArray(iter)[0];
 
-        NPY_BEGIN_THREADS_NDITER(iter);
+        PyArray_Descr **dts = NpyIter_GetDescrArray(iter);
+        PyArray_Descr *dtx = dts[2];
+        PyArray_Descr *dty = dts[3];
+        npy_intp itemsize = dts[0]->elsize;
+
+        npy_intp *strides = NpyIter_GetInnerStrideArray(iter);
+        npy_intp cstride = strides[1];
+        npy_intp xstride = strides[2];
+        npy_intp ystride = strides[3];
+
+        int axswap = PyDataType_ISBYTESWAPPED(dtx);
+        int ayswap = PyDataType_ISBYTESWAPPED(dty);
+        int native = (axswap == ayswap) && (axswap == 0) && !needs_api;
+
+        NPY_ARRAYMETHOD_FLAGS transfer_flags = 0;
+
+        npy_intp transfer_strides[2] = {xstride, itemsize};
```

Comment:
```suggestion
        npy_intp transfer_strides[2] = {itemsize, itemsize};
```
We copy only a single item anyway, so it doesn't matter.